docs/source/basics/prompt_templates.rst

```diff
@@ -0,0 +1,249 @@
+
+    System: {content}\\n
+    User: {content}\\n
+    Assistant: {content}\\n
+    Tool: {content}\\n
+
+You need to pass in a tuple for each role, where ``PREPEND_TAG`` is the string
+added before the text content and ``APPEND_TAG`` is the string added after.
+
+.. code-block:: python
+
+    template = {role: (PREPEND_TAG, APPEND_TAG)}
+
+Thus, the template would be defined as follows:
+
+.. code-block:: python
+
+    template = {
+        "system": ("System: ", "\\n"),
+        "user": ("User: ", "\\n"),
+        "assistant": ("Assistant: ", "\\n"),
+        "ipython": ("Tool: ", "\\n"),
+    }
+
+Now we can pass it into the tokenizer as a dictionary:
+
+.. code-block:: python
+
+    # In code
+    from torchtune.models.mistral import mistral_tokenizer
+
+    template = {
+        "system": ("System: ", "\\n"),
+        "user": ("User: ", "\\n"),
+        "assistant": ("Assistant: ", "\\n"),
+        "ipython": ("Tool: ", "\\n"),
+    }
+    m_tokenizer = mistral_tokenizer(
+        path="/tmp/Mistral-7B-v0.1/tokenizer.model"
+        prompt_template=template,
+    )
+
+.. code-block:: yaml
+
+    # In config
+    tokenizer:
+      _component_: torchtune.models.mistral.mistral_tokenizer
+      path: /tmp/Mistral-7B-v0.1/tokenizer.model
+      prompt_template:
+        system:
+          - "System: "
+          - "\\n"
+        user:
+          - "User: "
+          - "\\n"
+        assistant:
+          - "Assistant: "
+          - "\\n"
+        ipython:
+          - "Tool: "
+          - "\\n"
+
+If you don't want to add a prepend/append tag to a role, you can just pass in an empty string "" where needed.
+
+Using the :class:`~torchtune.data.PromptTemplate` class
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A template dictionary can also be passed into :class:`~torchtune.data.PromptTemplate` so you can use it as a standalone custom
+prompt template class.
+
+.. code-block:: python
+
+    from torchtune.data import PromptTemplate
+    from functools import partial
+
+    # As a builder function
+    def my_custom_template() -> PromptTemplate:
+        return PromptTemplate(
+            template={
+                "user": ("User: ", "\\n"),
+                "assistant": ("Assistant: ", "\\n"),
+            },
+        )
+
+    # As a class
+    MyCustomTemplate = partial(
+        PromptTemplate,
+        template={
+            "user": ("User: ", "\\n"),
+            "assistant": ("Assistant: ", "\\n"),
+        },
+    )
+
+    template = MyCustomTemplate()
+    msgs = [
+        Message(role="user", content="Hello world!"),
+        Message(role="assistant", content="Is AI overhyped?"),
+    ]
+    templated_msgs = template(msgs)
+    for msg in templated_msgs:
+        print(msg.role, msg.text_content)
+    # user, User: Hello world!
+    #
+    # assistant, Assistant: Is AI overhyped?
+    #
+
+.. TODO (RdoubleA) add a section on how to define prompt templates for inference once generate script is finalized
```

Comment:
Ya was just gonna comment that looking in the next section my first q is "what's up with the inference flag?" In the meantime maybe we can at least give some cursory sentence about it below?